src/lib.rs

```diff
@@ -1149,6 +1149,49 @@ fn graph_adjacency_matrix(
     Ok(matrix.into_pyarray(py).into())
 }
 
+#[pyfunction(cutoff=-1)]
+fn graph_all_simple_paths(
+    graph: &graph::PyGraph,
+    from: usize,
+    to: usize,
+    cutoff: isize,
+) -> PyResult<Vec<Vec<usize>>> {
+    let from_index = NodeIndex::new(from);
+    let to_index = NodeIndex::new(to);
+    let cutoff_petgraph: Option<usize>;
+    if cutoff < 1 {
+        cutoff_petgraph = None;
+    } else {
+        cutoff_petgraph = Some(cutoff as usize - 2);
+    }
+    let result: Vec<Vec<usize>> =
+        algo::all_simple_paths(graph, from_index, to_index, 0, cutoff_petgraph)
+            .map(|v: Vec<NodeIndex>| v.into_iter().map(|i| i.index()).collect())
+            .collect();
+    Ok(result)
+}
+
+#[pyfunction(cutoff=-1)]
+fn dag_all_simple_paths(
+    graph: &PyDAG,
+    from: usize,
+    to: usize,
+    cutoff: isize,
+) -> PyResult<Vec<Vec<usize>>> {
+    let from_index = NodeIndex::new(from);
+    let to_index = NodeIndex::new(to);
+    let cutoff_petgraph: Option<usize>;
+    if cutoff < 1 {
+        cutoff_petgraph = None;
+    } else {
+        cutoff_petgraph = Some(cutoff as usize - 2)
+    }
+    let result: Vec<Vec<usize>> =
+        algo::all_simple_paths(graph, from_index, to_index, 0, cutoff_petgraph)
```

Comment:
It might be nice to expose `min_intermediate_nodes` as well.

Reply:
Sure, I was just trying to mirror the networkx call which only has the cutoff parameter. But it's simple to add a kwarg to set this too. Should I do the same thing I did for cutoff and subtract 2 to make the python parameter depth?

Reply:
When I use this method I'm actually looking for all simply connected paths of length m on a device so this is useful. Depth seems good for the application I had in mind but maybe some algorithm could use the default too. Either is ok with me.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/85/commits/df4761ef36e9c117dfb084500a04583260412db3